src/Modules/Trinity.DynamicCluster/Trinity.DynamicCluster/ChunkedStorage.cs

```diff
@@ -2,52 +2,183 @@
         // route messages to a specific storage
-        List<Storage.Storage> Storages; // <----
-        Dictionary<Storage.Storage, ChunkCollection> Chunks; //<-----
-        MemoryCloud m_MemoryCloud; // get ChunkIds from m_MemoryCloud and decide routing.
+        private List<Storage.Storage> Storages; // <----
+        private Dictionary<Storage.Storage, ChunkCollection> Chunks; //<-----ChunkCollection
+        //DynamicMemoryCloud m_MemoryCloud; // get ChunkIds from m_MemoryCloud and decide routing.???????
+        //List<long> chunk_range = DynamicMemoryCloud.ChunkRange;
+        
+        internal TrinityErrorCode Mount(Storage.Storage s)
+        {
+            try
+            {
+                Storages.Add(s);
+                return TrinityErrorCode.E_SUCCESS;
+            }
+            catch (Exception ex)
+            {
+                Log.WriteLine(LogLevel.Error, "Errors occurred during Mount.");
+                Log.WriteLine(LogLevel.Error, ex.ToString());
+                return TrinityErrorCode.E_FAILURE;
+            }
+        }
 
-        bool ContainsLocalStorage { get { throw new NotFiniteNumberException(); } }
+        internal TrinityErrorCode UnMount(Storage.Storage s)
+        {
+            try
+            {
+                Storages.Remove(s);
+                return TrinityErrorCode.E_SUCCESS;
+            }
+            catch (Exception ex)
+            {
+                Log.WriteLine(LogLevel.Error, "Errors occurred during Mount.");
+                Log.WriteLine(LogLevel.Error, ex.ToString());
+                return TrinityErrorCode.E_FAILURE;
+            }
+        }
+
+        static internal long GetChunkIdByCellIdStatic(long cellId)
+        {
+            int chunk_id = -1;
+            for (chunk_id = 0; chunk_id < DynamicMemoryCloud.ChunkRange.Count; chunk_id++)
+            {
+                if (cellId > DynamicMemoryCloud.ChunkRange[chunk_id]) return chunk_id + 1;
+            }
+
+            return 0;
+        }
 
         public override TrinityErrorCode AddCell(long cellId, byte[] buff)
         {
-            throw new NotImplementedException();
+            foreach (Storage.Storage s in PickStorages(a => Chunks[a].Covers(cellId)))
+            {
+                try
+                {
+                    s.AddCell(cellId, buff);
+                }
+                catch (Exception ex)
+                {
+                    Log.WriteLine(LogLevel.Error, "Errors occurred during AddCell.");
+                    Log.WriteLine(LogLevel.Error, ex.ToString());
+                    return TrinityErrorCode.E_FAILURE;
+                }
+            }
+            return TrinityErrorCode.E_SUCCESS;
         }
 
         public override TrinityErrorCode AddCell(long cellId, byte* buff, int size)
         {
-            throw new NotImplementedException();
+            foreach (Storage.Storage s in PickStorages(a => Chunks[a].Covers(cellId)))
```

Comment:
Consider implement `_AddCell_impl(.......)` once, marked with `AggressiveInlining`, and then call it from every `AddCell` overload.